pkg/kubectl/kubectl.go

```diff
@@ -2,34 +2,14 @@ package kubectl
 
 import (
 	"fmt"
-	"os/exec"
 	"strings"
 )
 
-func executeCommand(args []string) (string, error) {
-	cmdOut, err := exec.Command("kubectl", args...).CombinedOutput()
-	if err != nil {
-		// Kubectl error messages output to stdOut
-		return "", fmt.Errorf(formatCmdOutput(cmdOut))
-	}
-	return formatCmdOutput(cmdOut), nil
-}
-
-// Execute executes kubectl <args> and returns the combined stdout/err output.
-func Execute(args ...string) (string, error) {
-	return executeCommand(args)
-}
-
-// ExecuteWithGlobalArgs is a convenience version of Execute that lets the user
-// specify global arguments as an array. Global arguments are arguments that are
-// not specific to a kubectl sub-command, eg. --kubeconfig. The list of global
-// options can be retrieved with kubectl options.
-func ExecuteWithGlobalArgs(globalArgs []string, args ...string) (string, error) {
-	return executeCommand(append(globalArgs, args...))
-}
-
-func formatCmdOutput(output []byte) string {
-	return strings.TrimPrefix(strings.TrimSuffix(strings.TrimSpace(string(output)), "'"), "'")
+// Client implements a kubectl client to execute commands
+type Client interface {
+	Execute(args ...string) (string, error)
```

Comment:
Totally optional, I don't know too well the requirement on the gcp service side:

Execute() and ExcuteWithGlobalArgs() are really the same thing. The interface should probably just implement the lower-common denominator and Execute()/ExecuteWithGlobalArgs() implemented in terms of that. In the local implementation this is Execute(args []string). We could then implement the two methods as:

```
package kubectl

func Execute() (k KubectlClient, args ...string) {
    k.Execute(args...)
}

func ExecuteWithGlobalArgs() (k KubectlClient, globalArgs []string, args ...string) {
     k.Execute(append(globalArgs, args...))
}
```